apps/angular/ngfor-enhancement/src/app/app.component.ts

```diff
@@ -1,25 +1,33 @@
-import { NgFor, NgIf } from '@angular/common';
 import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { NgForEmptyDirective } from './ng-for-empty.directive';
 
 interface Person {
   name: string;
 }
 
 @Component({
   standalone: true,
-  imports: [NgFor, NgIf],
+  imports: [NgForEmptyDirective],
   selector: 'app-root',
   template: `
-    <ng-container *ngIf="persons.length > 0; else emptyList">
-      <div *ngFor="let person of persons">
-        {{ person.name }}
-      </div>
-    </ng-container>
+    <div><button (click)="addPerson()">Add</button></div>
+    <div *ngForEmpty="let person of persons; empty: emptyList">
+      {{ person.name }}
+      <button (click)="removePerson(person)">Remove</button>
+    </div>
     <ng-template #emptyList>The list is empty !!</ng-template>
   `,
   styles: [],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AppComponent {
   persons: Person[] = [];
```

Comment:
question: why not a signal here ? With zoneless coming, you should transform everything into signals.

Reply:
You're totally right. I just didn't think of it.